File: ios/RoutineChart/RoutineChart/Core/Domain/Repositories/FamilyInviteRepository.swift
//
//  FamilyInviteRepository.swift
//  RoutineChart
//
//  Created for Phase 2.2: QR Family Joining
//

import Foundation

protocol FamilyInviteRepository {
    /// Create a new family invite
    func create(_ invite: FamilyInvite) async throws
    
    /// Get invite by ID
    func get(id: String) async throws -> FamilyInvite?
    
    /// Get invite by token
    func getByToken(_ token: String) async throws -> FamilyInvite?
    
    /// Get invite by invite code (e.g., "ABC-1234")
    func getByInviteCode(_ inviteCode: String) async throws -> FamilyInvite?
    
    /// Get all active invites for a family
    func getActiveInvites(familyId: String) async throws -> [FamilyInvite]
    
    /// Update an invite (for incrementing usedCount or deactivating)
    func update(_ invite: FamilyInvite) async throws
    
    /// Deactivate an invite
    func deactivate(id: String) async throws
    
    /// Delete expired invites
    func deleteExpired() async throws
}

File: android/app/src/main/java/com/HammersTech/RoutineChart/core/domain/repositories/FamilyInviteRepository.kt
package com.HammersTech.RoutineChart.core.domain.repositories

import com.HammersTech.RoutineChart.core.domain.models.FamilyInvite

/**
 * Repository interface for family invites
 * Phase 2.2: QR Family Joining
 */
interface FamilyInviteRepository {
    /**
     * Create a new family invite
     */
    suspend fun create(invite: FamilyInvite)

    /**
     * Get invite by ID
     */
    suspend fun getById(id: String): FamilyInvite?

    /**
     * Get invite by token
     */
    suspend fun getByToken(token: String): FamilyInvite?

    /**
     * Get invite by invite code (e.g., "ABC-1234")
     */
    suspend fun getByInviteCode(inviteCode: String): FamilyInvite?

    /**
     * Get all active invites for a family
     */
    suspend fun getActiveInvites(familyId: String): List<FamilyInvite>

    /**
     * Update an invite (for incrementing usedCount or deactivating)
     */
    suspend fun update(invite: FamilyInvite)

    /**
     * Deactivate an invite
     */
    suspend fun deactivate(id: String)

    /**
     * Delete expired invites
     */
    suspend fun deleteExpired()
}
